### PR TITLE
Blog post storage proxy 405

### PR DESCRIPTION
**Pull request by Aliamondo**
Need image suggestions for header image.
Do tell if I need to fix wording in the post.

### CODE REVIEW

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+Searching for similar issues, trying out the solutions from all the different posts on Github and Stackoverflow, failing to run them successfully, searching for new solutions, etc.
+Nothing helped.
+My team tried to lend me a hand and after every discussion I would go and make a change in the microservice, only to find out that the changes did not solve the problem.
+
+After a certain point I did not believe that we can upgrade our proxy microservice in the first place and felt like giving up on the task altogether (which means putting it back to the backlog).
+This was when Donald told me he can take this problem over.
+
+---
+
+Integration tests, Distributed Tracing and Logging are the terms that we developers tend to ignore or not worry that much since they don't appear in the scene on sunny days.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:106]: Suggestion: "Integration tests", "Distributed tracing" and "Logging" are the terms...

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+But working on a microservice architecture where quite often you have to debug requests passing through several services, the lack of traceability or a centralized logging system can be quite a hassle.
+Luckily our logs are analyzed and stored to `Google Stackdriver` which offers you plenty of functionalities to search through the logs.
+
+Let's go back to the problem and to what I did to find it and come up with the solution in the end.
+
+After discussing with Andrey what had he tried and I decide to go Sherlock and follow the failing requests around in order to see what was changed on its way.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:112]: After discussing with Andrey what he had tried, I decided to go Sherlock and follow the failing requests around in order to see what was changed on its way.

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+{% image_custom image="/assets/img/pages/blog/images/sherlock.png" width="80" lightbox %}
+
+As it's know every request in a Spring based environment goes through a chain of filters which can validate, enhance or even block the requests.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:116]: As it is known, every request in a Spring based environment goes through a chain of filters which can validate, enhance, or even block the requests.

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+Same like passing the security checks at the airport!
+
+Thus, I went through each filter and everything was look fine until I reached the recently added `ForwardedHeaderFilter`.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:119]: everything was **looking** fine

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+Finally, a clue appeared.
+The request path was changed to a different path which was a bit unfortunate for us since it existed only for `POST` requests therefor a `405` was received.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:121]: I don't understand this part: "...therefore a `405` was received."

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+After reading the documentation I understood that it's used to wrap the initial request and change it based on the `X-Forwarded` headers, a behavior which we didn't want to have.
+Then **WHY?** Why an unnecessary filter was added on the first place? Answer is simple.
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:123]: Then **WHY** an unnecessary filter was added on the first place? The answer is simple.

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+And now the vicious loop begins.
+Searching for similar issues, trying out the solutions from all the different posts on Github and Stackoverflow, failing to run them successfully, searching for new solutions, etc.
+Nothing helped.
+My team tried to lend me a hand and after every discussion I would go and make a change in the microservice, only to find out that the changes did not solve the problem.
+
+After a certain point I did not believe that we can upgrade our proxy microservice in the first place and felt like giving up on the task altogether (which means putting it back to the backlog).
+This was when Donald told me he can take this problem over.
+
+---
```
Comment (amahaiuddin) [inline _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md:104]: Please add here a headline for the following part.

File: _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md (hunk 5 above)
Comment (Aliamondo) [reply _posts/2020/2020-08-10-spring-boot-upgrade-not-allowed.md]: :man_shrugging: 
I tried to fix it

File: _posts/2020/2020-08-11-why-our-spring-boot-upgrade-was-not-allowed.md [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+
+It's Bean time!
+```java
+@Bean
+public Filter forwardedHeaderFilter() {
+    return new ForwardedHeaderFilter();
+}
+```
+
+With this bean in configuration class, I ran the tests again, and they all successfully passed.
```
Comment (cgebken) [inline _posts/2020/2020-08-11-why-our-spring-boot-upgrade-was-not-allowed.md:86]: Do you have a link we could add here for more information on bean?

Comment (Aliamondo) [reply _posts/2020/2020-08-11-why-our-spring-boot-upgrade-was-not-allowed.md:86]: The general information on what Beans are? Or what this specific Bean is?

https://docs.spring.io/spring/docs/4.3.14.RELEASE_to_4.3.15.RELEASE/Spring%20Framework%204.3.15.RELEASE/org/springframework/web/filter/ForwardedHeaderFilter.html

Comment (cgebken) [reply _posts/2020/2020-08-11-why-our-spring-boot-upgrade-was-not-allowed.md:86]: Ah okay I thought that these might also be a kind of class or method that have their own documentation. But if there's only the general one, I'm fine with not linking it :)

**Pull request merged**